packages/core/src/odata-common/entity-builder.spec.ts

```diff
@@ -38,40 +40,145 @@ describe('EntityBuilder', () => {
     ).toEqual(expected);
   });
 
-  it('should build an entity from json', () => {
-    const builder = TestEntity.builder();
-    const linkedEntity = TestEntitySingleLink.builder()
-      .stringProperty('someString')
-      .withCustomFields({ linkedCustomField: 'someLinkedValue' })
-      .build();
-
-    const entityJson: Partial<TestEntity> = {
-      stringProperty: 'someValue',
-      toSingleLink: linkedEntity
-    };
-    entityJson['_customFields'] = { additionalField: 'someAdditionalValue' };
-    const actual = builder.fromJson(entityJson);
-    expect(actual.stringProperty).toBe(entityJson.stringProperty);
-    expect(actual.getCustomFields()).toEqual(entityJson['_customFields']);
-    expect(actual.toSingleLink.stringProperty).toBe(
-      linkedEntity.stringProperty
-    );
-    expect(actual.toSingleLink.getCustomFields()).toEqual(
-      linkedEntity.getCustomFields()
-    );
-  });
-
   it('should build an entity with non-primitive JS types (moment, BigNumber etc.)', () => {
     const expected: TestEntity = new TestEntity();
 
-    expected.dateTimeProperty = moment();
+    expected.dateProperty = moment();
     expected.decimalProperty = new BigNumber(10);
 
     const actual = TestEntity.builder()
-      .dateTimeProperty(expected.dateTimeProperty)
+      .dateProperty(expected.dateProperty)
       .decimalProperty(expected.decimalProperty)
       .build();
 
     expect(expected).toEqual(actual);
   });
+
+  describe('fromJson', () => {
```

Comment:
The `fromJson` method is called recursivly via the `buildNavigationPropertyFromJson` could you also add a test for this case in other words at least two levels of nested json properties.

Reply:
What case would be covered, that currently isn't?
Exactly because the function is called recursively, it is called again and therefore if it works on the first level, it works on all levels. Maybe I am missing something?

Reply:
Will add as discussed

Reply:
Done